### PR TITLE
CB-21-stock-api

### PR DESCRIPTION
**Pull request by DHua5922**
* **Please check if the PR fulfills these requirements**
- [X] The pull request title contains the ticket number in the title. For example [CB-1]

* **What kind of change does this PR introduce?** (Bug fix, new feature...)

New feature: API for getting stock quote.

* **What is the current behavior?What does the code do before you made the new changes?** 

No API for getting stock quote.

* **What is the new behavior?**

Call API with `/quote` to get stock quote.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR? Run a new command?)

Install [finnhub npm package](https://www.npmjs.com/package/finnhub) with `npm i finnhub`.

* **Other information**:

### CODE REVIEW

**Comment by sunnyguan** [comment]
Is the stock API key the same for #13? If so, I can change my PR so that we only need one environment variable `API_KEY`. I'll change #13 to use the Finnhub JS API too so everything is cohesive.

File: backend/index.js [hunk 1]
```diff
@@ -14,4 +18,12 @@ app.listen(port,()=>{
     console.log('API is up and running')
 })
 
+//Quote
+app.get('/quote', (req, res) => {
+    finnhubClient.quote("AAPL", (error, data, response) => {
```
Comment (sunnyguan) [inline backend/index.js:23]: Maybe add in a URL parameter so it's easier to integrate with the front end: `var company = req.query.company;`

**Comment by DHua5922** [comment]
> Is the stock API key the same for #13? If so, I can change my PR so that we only need one environment variable `API_KEY`. I'll change #13 to use the Finnhub JS API too so everything is cohesive.
> 
> Edit: Since a lot of the Finnhub API code is the same for stock and news calls, I'll merge #13 with main after this PR is approved & merged.

I think the stock API key is the same for #13.

File: backend/index.js (hunk 1 above)
Comment (Codubee) [reply backend/index.js:23]: I agree with Sunny here.

File: backend/index.js [hunk 2]
```diff
@@ -1,11 +1,15 @@
 const express = require('express');
 const app = express();
-const axios = require('axios');
 require('dotenv').config();
 app.use(express.json())
 
 const port = 8080;
 
+const finnhub = require('finnhub');
```
Comment (Codubee) [inline backend/index.js:8]: Nice! Making us of the NPM module.

File: backend/index.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+const api_key = finnhub.ApiClient.instance.authentications['api_key'];
+api_key.apiKey = process.env.API_KEY;
```
Comment (Codubee) [inline backend/index.js:10]: Can we name the key here something more descriptive. If we ever add more keys to the .env file then we need a good way to now which api key goes to which api.

**Review by Codubee — CHANGES_REQUESTED** [review]
Hey Dylan I left a couple of comments. Also Sunnys comments.

**Review by sunnyguan — APPROVED** [review]
Looks good to me, `/quote` is working as expected

**Review by Codubee — APPROVED** [review]
LGTM

**Pull request merged**